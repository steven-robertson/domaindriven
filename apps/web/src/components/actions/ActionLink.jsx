import React from "react";
import PropTypes from "prop-types";

ActionLink.propTypes = {
    onClick: PropTypes.func.isRequired,
    children: PropTypes.string.isRequired
}

export default function ActionLink(props) {
    return <span className="action-link" onClick={props.onClick}>{props.children}</span>
}
