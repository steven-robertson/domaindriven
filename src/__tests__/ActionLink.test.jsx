import React from "react";
import renderer from "react-test-renderer";
import ActionLink from "../components/actions/ActionLink";

it('matches snapshot', () => {
    const component = renderer.create(
        <ActionLink onClick={() => {}}>
            Name
        </ActionLink>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
