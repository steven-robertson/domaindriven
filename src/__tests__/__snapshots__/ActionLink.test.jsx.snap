// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  className="action-link"
  onClick={[Function]}
>
  Name
</span>
`;
